sdk/core/azure-core/src/main/java/com/azure/core/implementation/util/EnvironmentConfiguration.java

```diff
@@ -0,0 +1,241 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.implementation.util;
+
+import com.azure.core.util.Configuration;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.function.Function;
+
+/**
+ * Contains environment (system properties and environment variables) configuration information that is
+ * used during construction of client libraries.
+ */
+public class EnvironmentConfiguration implements Cloneable {
```

Comment:
We should generally avoid implementing `Cloneable` interface. We should instead use copy constructor if we want to clone another instance of EnvironmentConfiguration.